### PR TITLE
Added insert element by index and clearList to DoublyLinkedList.java

### PR DESCRIPTION
**Pull request by rnitish**
I have added two methods:

1. insertElementByIndex - add an element to specified position.
2. clearList - clears the DoublyLinkedList.

Fixed some errors

closes issue #1378

### CODE REVIEW

File: DataStructures/Lists/DoublyLinkedList.java [hunk 1]
```diff
@@ -43,6 +49,7 @@ public DoublyLinkedList(int[] array) {
         for (int i : array) {
             insertTail(i);
         }
+        size = array.length;
```
Comment (rbshealy) [inline DataStructures/Lists/DoublyLinkedList.java:52]: If size > 0 before this function is called, size != array.length

Comment (rbshealy) [reply DataStructures/Lists/DoublyLinkedList.java:52]: use size += array.length

Comment (rnitish) [reply DataStructures/Lists/DoublyLinkedList.java:52]: Thank you @rbshealy  for reviewing, I will make the required changes.

Comment (rbshealy) [reply DataStructures/Lists/DoublyLinkedList.java:52]: Yes this is correct, didn't realize this is the constructor. Great work!

Comment (rnitish) [reply DataStructures/Lists/DoublyLinkedList.java:52]: Thank you

**Pull request merged**